bentoml/_internal/bento/gen.py

```diff
@@ -115,71 +110,99 @@ def get_template_env(
                 python_version = python_version
             base_image = spec.image.format(spec_version=python_version)
     else:
-        base_image = docker_options.base_image
+        base_image = options.base_image
         logger.info(
             f"BentoML will not install Python to custom base images; ensure the base image '{base_image}' has Python installed."
         )
 
+    # environment returns are
+    # __base_image__, __supported_architectures__, __bentoml_version__, __python_version_full__
+    # bento__uid_gid, bento__user, bento__home, bento__path
+    # __options__distros, __options__base_image, __options_env, __options_system_packages, __options_setup_script
     return {
-        **{
-            f"__options__{k}": v
-            for k, v in bentoml_cattr.unstructure(docker_options).items()
-        },
-        **bentoml_cattr.unstructure(CustomizableEnv()),
-        **bentoml_cattr.unstructure(ReservedEnv(base_image, supported_architectures)),
+        **{f"__options__{k}": v for k, v in attr.asdict(options).items()},
+        **CustomizableEnv().todict(),
+        **ReservedEnv(base_image, spec.supported_architectures).todict(),
```

Comment:
when base image is coming from `base_image = options.base_image`, there's no spec.supported_architectures, right?